src/test/java/io/vertx/test/fakecluster/FakeClusterManager.java

```diff
@@ -207,107 +203,162 @@ public void leave(Handler<AsyncResult<Void>> resultHandler) {
         }
       }
       fut.complete();
-    }, resultHandler);
+    });
   }
 
   @Override
   public boolean isActive() {
     return nodeID != null;
   }
 
+  @Override
+  public Future<Void> register(String address, RegistrationInfo registrationInfo) {
+    registrations.computeIfAbsent(address, k -> Collections.synchronizedList(new ArrayList<>()))
+      .add(registrationInfo);
+    return vertx.getOrCreateContext().succeededFuture();
+  }
+
+  @Override
+  public Future<Void> unregister(String address, RegistrationInfo registrationInfo) {
+    Future<Void> result;
+    List<RegistrationInfo> infos = registrations.get(address);
```

Comment:
instead use `ConcurrentMap#compute` where you can return `null` to remove the entry. The current implementation is not race free, i.e you can have concurrent removal and addition that will let an orphan registration list in the map or a non empty registration list to be not registered

Reply:
I'm not sure to understand, could you give an example to illustrate the issue?